samples/MQTTSN/src/paho_iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,280 @@
+    return rc;
+  }
+
+  options.clientID.cstring = device_id;
+
+  // Get topic name that the IoT Hub is subscribed to
+  if (az_failed(
+          rc = az_iot_hub_client_telemetry_get_publish_topic(
+              &client, NULL, topicname, sizeof(topicname), NULL)))
+  {
+    printf("Failed to get publish topic, return code %d\r\n", rc);
+    return rc;
+  }
+
+  // CONNECT to MQTTSN Gateway
+  len = MQTTSNSerialize_connect(buf, buflen, &options);
+
+  if (az_failed(rc = transport_sendPacketBuffer(host, port, buf, len)))
+  {
+    printf("Failed to send Connect packet to the Gateway, return code %d\r\n", rc);
+    return rc;
+  }
+
+  // Wait for CONNACK from the MQTTSN Gateway
+  if (MQTTSNPacket_read(buf, buflen, transport_getdata) == MQTTSN_CONNACK)
+  {
+    int connack_rc = -1;
+
+    if (MQTTSNDeserialize_connack(&connack_rc, buf, buflen) != 1 || connack_rc != 0)
+    {
+      printf("Failed to receive Connect ACK packet, return code %d\r\nExiting...\r\n", connack_rc);
+      goto exit;
+    }
+    else
+      printf("CONNACK rc %d\r\n", connack_rc);
+  }
+  else
+  {
+    printf("Failed to connect to the Gateway\r\nExiting...\r\n");
+    goto exit;
+  }
+
+  // REGISTER topic name with the MQTTSN Gateway
+  printf("Registering topic %s\r\n", topicname);
+  topicstr.cstring = topicname;
+  topicstr.lenstring.len = strlen(topicname);
+  len = MQTTSNSerialize_register(buf, buflen, 0, packetid, &topicstr);
+
+  if (az_failed(rc = transport_sendPacketBuffer(host, port, buf, len)))
+  {
+    printf("Failed to send Register packet to the Gateway, return code %d\r\n", rc);
+    return rc;
+  }
+
+  // Wait for REGACK from the MQTTSN Gateway
+  if (MQTTSNPacket_read(buf, buflen, transport_getdata) == MQTTSN_REGACK)
+  {
+    unsigned short submsgid;
+    unsigned char returncode;
+
+    rc = MQTTSNDeserialize_regack(&topicid, &submsgid, &returncode, buf, buflen);
+    if (returncode != 0)
+    {
+      printf("Failed to receive Register ACK packet, return code %d\r\nExiting...\r\n", returncode);
+      goto exit;
+    }
+    else
+      printf("REGACK topic id %d\r\n", topicid);
+  }
+  else
+  {
+    printf("Failed to register topic with the Gateway\r\nExiting...\r\n");
+    goto exit;
+  }
+
+  // Publish 5 messages
+  for (int i = 0; i < NUMBER_OF_MESSAGES; ++i)
+  {
+    printf("Sending Message %d\r\n", i + 1);
+    topic.type = MQTTSN_TOPIC_TYPE_NORMAL;
+    topic.data.id = topicid;
+
+    // PUBLISH
+    len = MQTTSNSerialize_publish(
+        buf,
+        buflen,
+        0,
+        QOS,
+        retained,
+        packetid + i,
+        topic,
+        TELEMETRY_PAYLOAD,
+        sizeof(TELEMETRY_PAYLOAD));
+
+    if (az_failed(rc = transport_sendPacketBuffer(host, port, buf, len)))
+    {
+      printf("Failed to publish telemetry message %d, return code %d\r\n", i + 1, rc);
+      return rc;
+    }
+
+    printf("Published rc %d for publish length %d\r\n", rc, len);
+
+    if (QOS == 1)

```

Comment:
This might not be an optimal way of doing things.

Lets try to first create functions and then wrap function with #ifdef so that is not getting used they are not part of the binary.